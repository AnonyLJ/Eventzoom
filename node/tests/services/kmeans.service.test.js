import kmeansService from '../../src/services/kmeans.service';

describe('calculate', () => {
	it('should calculate', async () => {
		const compare = (num1, num2) => Math.abs(num1 - num2);
		const calculateCentroids = (nums) => nums[0];
		expect(kmeansService.calculate([1, 2, 3, 4, 5, 6, 7], [1, 2], compare, calculateCentroids, 500))
			.toMatchSnapshot();
	});
});
