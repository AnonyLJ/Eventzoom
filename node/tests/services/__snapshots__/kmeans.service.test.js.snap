// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate should calculate 1`] = `
Array [
  Array [
    1,
  ],
  Array [
    2,
    3,
    4,
    5,
    6,
    7,
  ],
]
`;
